packages/dds/tree/src/test/shared-tree/sharedTreeFuzzTests.ts

```diff
@@ -0,0 +1,287 @@
+                await testTreeProvider.ensureSynchronized();
+                return state;
+            },
+        },
+        initialState,
+        saveInfo,
+    );
+
+    await finalState.testTreeProvider.ensureSynchronized();
+    validateTree(provider.trees[0], [initialTreeState]);
+
+    if (validateAnchor) {
+        const expectedPath: UpPath = {
+            parent: {
+                parent: undefined,
+                parentIndex: 0,
+                parentField: rootFieldKeySymbol,
+            },
+            parentField: brand("foo"),
+            parentIndex: 1,
+        };
+        const anchorPath = tree.locate(firstAnchor);
+        assert(compareUpPaths(expectedPath, anchorPath));
+    }
+    return finalState;
+}
+
+function applyFuzzChange(tree: ISharedTree, contents: FuzzChange, transactionResult:TransactionResult): void {
+    const index = contents.index;
+    const nodeField = contents.field;
+    switch (contents.fuzzType) {
+        case "insert":
+            if (index !== undefined && nodeField !== undefined) {
+                tree.runTransaction((forest, editor) => {
+                    const field = editor.sequenceField(contents.parent, nodeField);
+                    field.insert(
+                        index,
+                        singleTextCursor({ type: brand("Test"), value: contents.value }),
+                    );
+                    return transactionResult;
+                });
+            }
+            break;
+        case "delete":
+            if (index !== undefined && nodeField !== undefined) {
+                const parent = contents.parent?.parent;
+                const delField = contents.parent?.parentField;
+                const parentIndex = contents.parent?.parentIndex;
+                if (delField !== undefined && parent !== undefined && parentIndex !== undefined) {
+                    tree.runTransaction((forest, editor) => {
+                        const field = editor.sequenceField(parent, delField);
+                        field.delete(parentIndex, 1);
+                        return transactionResult;
+                    });
+                }
+            }
+            break;
+        case "setPayload":
+            if (index !== undefined && nodeField !== undefined) {
+                const path = contents.parent;
+                if (path !== undefined) {
+                    tree.runTransaction((forest, editor) => {
+                        editor.setValue(path, contents.value);
+                        return transactionResult;
+                    });
+                }
+            }
+            break;
+        default:
+            fail("Invalid edit.");
+    }
+}
+
+enum FuzzTestType {
+    Basic = 0,
+    Abort = 1,
+    AnchorStability = 2,
+}
+
+export function runSharedTreeFuzzTests(title: string): void {
+    const random = makeRandom(0)
+    const testBatchSize = 3;
+    describeFuzz(title, ({ testCount }) => {
+        function runFuzzTest (
```

Comment:
Declaring the function here reduces readability in two ways:
1. The code structure is just harder to see because the code for the outer function gets interupted by the code for the inner one.
2. It's not obvious what the inner function captures from the outer function's scope (it looks like `random` might be the only thing)

Please pull this function out into the module scope and update its parameters as needed.
I think renaming it to something like `runBatch` would also make it clearer what this function's reponsibility is (the "fuzz" part can be inferred from the module we're in (we'd want to keep the "fuzz" part of the name if the function were exported though)).